src/khoj/routers/api_chat.py

```diff
@@ -526,141 +521,142 @@ async def set_conversation_title(
     )
 
 
-@api_chat.websocket("/ws")
-async def websocket_endpoint(
-    websocket: WebSocket,
-    conversation_id: int,
+@api_chat.get("")
+async def chat(
+    request: Request,
+    common: CommonQueryParams,
+    q: str,
+    n: int = 7,
+    d: float = 0.18,
+    stream: Optional[bool] = False,
+    title: Optional[str] = None,
+    conversation_id: Optional[int] = None,
     city: Optional[str] = None,
     region: Optional[str] = None,
     country: Optional[str] = None,
     timezone: Optional[str] = None,
+    rate_limiter_per_minute=Depends(
+        ApiUserRateLimiter(requests=5, subscribed_requests=60, window=60, slug="chat_minute")
+    ),
+    rate_limiter_per_day=Depends(
+        ApiUserRateLimiter(requests=5, subscribed_requests=600, window=60 * 60 * 24, slug="chat_day")
+    ),
 ):
-    connection_alive = True
-
-    async def send_status_update(message: str):
-        nonlocal connection_alive
-        if not connection_alive:
-            return
+    async def event_generator(q: str):
+        start_time = time.perf_counter()
+        ttft = None
+        chat_metadata: dict = {}
+        connection_alive = True
+        user: KhojUser = request.user.object
+        event_delimiter = "␃🔚␗"
```

Comment:
I haven't understood why we need this. Wouldn't it be sufficient for the client to try parsing the message as a JSON response, and if it is not a JSON response, then assume that it's the LLM stream?

Reply:
A single message chunk can have partial JSON or multiple JSONs. For both those scenarios the above suggested mechanism would fail. The message chunking is decided by the server and is variable. So relying on a special delimiter helps us avoid parsing status messages on the clients